package.nix

```diff
@@ -0,0 +1,31 @@
+{ pkgs, inputs }:
+
+let
+  python_slim = pkgs.python311.override {
```

Comment:
I think the problems you're seeing is related to how the package set is injected onto the derivation.
You also need to pass `self`:
 ``` nix
python_slim = pkgs.python311.override {
  self = python_slim
}
```

Reply:
Package set injection in poetry or nixpkgs?

Reply:
In nixpkgs

Reply:
Managed to fix it, thanks!